From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Sun, 25 Dec 2022 10:47:29 +0100
Subject: [PATCH] Skip secondary POI sensor if absent

License: LGPL-3.0-only (https://www.gnu.org/licenses/lgpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following mixin:
"net/caffeinemc/mods/lithium/mixin/ai/sensor/secondary_poi/SecondaryPoiSensorMixin.java"
By: 2No2Name <2No2Name@web.de>
As part of: Lithium (https://github.com/CaffeineMC/lithium)
Licensed under: LGPL-3.0-only (https://www.gnu.org/licenses/lgpl-3.0.html)

* Lithium description *

villagers that have no secondary POI (farmland) do not search for it

diff --git a/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java b/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
index 1da9f1cdb2ee8b1514c79fee24460f852b6a3577..3c0b272ae11df3b2f929a55497278bd9bad51422 100644
--- a/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
@@ -24,7 +24,10 @@ public class SecondaryPoiSensor extends Sensor<Villager> {
     protected void doTick(final ServerLevel level, final Villager body) {
         // Purpur start - Option for Villager Clerics to farm Nether Wart - make sure clerics don't wander to soul sand when the option is off
         Brain<?> brain = body.getBrain();
-        if (!level.purpurConfig.villagerClericsFarmWarts && body.getVillagerData().profession().is(net.minecraft.world.entity.npc.villager.VillagerProfession.CLERIC)) {
+        // Gale start - Lithium - skip secondary POI sensor if absent
+        var secondaryPoi = body.getVillagerData().profession().value().secondaryPoi();
+        if (secondaryPoi.isEmpty() || (!level.purpurConfig.villagerClericsFarmWarts && body.getVillagerData().profession().is(net.minecraft.world.entity.npc.villager.VillagerProfession.CLERIC))) {
+            // Gale end - Lithium - skip secondary POI sensor if absent
             brain.eraseMemory(MemoryModuleType.SECONDARY_JOB_SITE);
             return;
         }
@@ -39,7 +42,7 @@ public class SecondaryPoiSensor extends Sensor<Villager> {
             for (int y = -2; y <= 2; y++) {
                 for (int z = -4; z <= 4; z++) {
                     BlockPos testPos = center.offset(x, y, z);
-                    if (body.getVillagerData().profession().value().secondaryPoi().contains(level.getBlockState(testPos).getBlock())) {
+                    if (secondaryPoi.contains(level.getBlockState(testPos).getBlock())) { // Gale - Lithium - skip secondary POI sensor if absent
                         jobSites.add(GlobalPos.of(dimensionType, testPos));
                     }
                 }
